frame/staking/src/migrations.rs

```diff
@@ -18,7 +18,85 @@
 
 use super::*;
 use frame_election_provider_support::SortedListProvider;
-use frame_support::traits::OnRuntimeUpgrade;
+use frame_support::{
+	dispatch::GetStorageVersion, pallet_prelude::ValueQuery, storage_alias,
+	traits::OnRuntimeUpgrade,
+};
+
+/// Outdated since v12. Useful to make encoding/decoding of old migration code easier.
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+enum ObsoleteReleases {
+	V1_0_0Ancient,
+	V2_0_0,
+	V3_0_0,
+	V4_0_0,
+	V5_0_0,  // blockable validators.
+	V6_0_0,  // removal of all storage associated with offchain phragmen.
+	V7_0_0,  // keep track of number of nominators / validators in map
+	V8_0_0,  // populate `VoterList`.
+	V9_0_0,  // inject validators into `VoterList` as well.
+	V10_0_0, // remove `EarliestUnappliedSlash`.
+	V11_0_0, // Move pallet storage prefix, e.g. BagsList -> VoterBagsList
+	V12_0_0, // remove `HistoryDepth`.
+}
+
+impl Default for ObsoleteReleases {
+	fn default() -> Self {
+		ObsoleteReleases::V12_0_0
+	}
+}
+
+/// Alias to the old storage item for keeping versions. Outdated since v12.
+#[storage_alias]
+type StorageVersion<T: Config> = StorageValue<Pallet<T>, ObsoleteReleases, ValueQuery>;
```

Comment:
Shouldn't this be scoped in v13?

Reply:
Its used in all migrations prior to v13. May be we should get rid of old migrations (unless you see a reason to keep them) and keep it scoped only in v13?